crowbar_framework/spec/controllers/api/upgrade_controller_spec.rb

```diff
@@ -57,6 +57,22 @@
       expect(response).to have_http_status(:unprocessable_entity)
     end
 
+    it "initiates the upgrade of nodes" do
```

Comment:
It's great that you're adding tests for this, but please can they be in the same commit which introduces the code they're testing?